gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/ContentValues.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import java.io.Serializable;
+import java.util.HashSet;
+import java.util.Set;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+
+/** Contains details about one content id: key and the expired values. */
+public class ContentValues implements Serializable {
+
+  private static final long serialVersionUID = -1268447510930923804L;
+  private ContentKey contentKey;
+  private final Set<Content> expiredContents = new HashSet<>();
+
+  public ContentKey getContentKey() {
+    return contentKey;
+  }
+
+  void gotValue(Content content, ContentKey key) {
+    synchronized (this) {
```

Comment:
Synchronziation here is applied in the wrong way. updates are protected, reads are not. That's inconsistent and dangerous. If it's intentional, it should clarify that via code (not just a comment).

Reply:
removed the synchronization as each reference we add contents sequentially.